extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -284,7 +284,15 @@ export class AmpAd3PImpl extends AMP.BaseElement {
             this.element);
         this.apiHandler_ = new AmpAdApiHandler(
           this, this.element, this.boundNoContentHandler_);
-        return this.apiHandler_.startUp(this.iframe_, true);
+        if (this.element.isFirstLayoutCompleted()) {
+          return this.apiHandler_.startUp(this.iframe_, true).then(() => {
```

Comment:
@zhouyx This is still being worked on, right? E.g. moving it to the `apiHandler`?